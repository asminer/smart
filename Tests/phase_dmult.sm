
// $Id$


print("=======================================\n");
print("Checking 3*equilikely(1,2)\n");
print("=======================================\n");
print_ph(equilikely(1,2));
print_ph(3*equilikely(1,2));

print("=======================================\n");
print("Checking 3*equilikely(0,2)\n");
print("=======================================\n");
print_ph(equilikely(0,2));
print_ph(3*equilikely(0,2));

print("=======================================\n");
print("Checking 5*geometric(0.2)\n");
print("=======================================\n");
print_ph(geometric(0.2));
print_ph(5*geometric(0.2));

print("=======================================\n");
print("Checking 5*(geometric(0.5)+geometric(0.2))\n");
print("=======================================\n");
print_ph(geometric(0.5)+geometric(0.2));
print("(Natural order)\n");
print_ph(5*(geometric(0.5)+geometric(0.2)));
print("(Lexical order)\n");
#StateDisplayOrder LEXICAL
#StatevectDisplayStyle SPARSE_STATE
print_ph(5*(geometric(0.5)+geometric(0.2)));
#StateDisplayOrder NATURAL
#StatevectDisplayStyle SPARSE_INDEX






