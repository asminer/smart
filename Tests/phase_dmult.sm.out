=======================================
Checking 3*equilikely(1,2)
=======================================
Information for phase type Equilikely(1, 2)
Discrete phase type, state has dimension 1
Reachable states:
State 0: 1
State 1: 2
State 2: a
3 states total
Accepting state index: 2
No trap state
Initial distribution:
    (0:1)
Markov chain:
Row 0:
	1 : 0.5
	2 : 0.5
Row 1:
	2 : 1
Row 2:
	2 : 1
3 edges total
End of information for phase type

Information for phase type 3*Equilikely(1, 2)
Discrete phase type, state has dimension 2
Reachable states:
State 0: [1, 3]
State 1: [1, 2]
State 2: [1, 1]
State 3: [2, 3]
State 4: [2, 2]
State 5: [2, 1]
State 6: [a, 1]
7 states total
Accepting state index: 6
No trap state
Initial distribution:
    (0:1)
Markov chain:
Row 0:
	1 : 1
Row 1:
	2 : 1
Row 2:
	3 : 0.5
	6 : 0.5
Row 3:
	4 : 1
Row 4:
	5 : 1
Row 5:
	6 : 1
Row 6:
	6 : 1
7 edges total
End of information for phase type

=======================================
Checking 3*equilikely(0,2)
=======================================
Information for phase type Equilikely(0, 2)
Discrete phase type, state has dimension 1
Reachable states:
State 0: 1
State 1: 2
State 2: a
3 states total
Accepting state index: 2
No trap state
Initial distribution:
    (0:0.666667, 2:0.333333)
Markov chain:
Row 0:
	1 : 0.5
	2 : 0.5
Row 1:
	2 : 1
Row 2:
	2 : 1
3 edges total
End of information for phase type

Information for phase type 3*Equilikely(0, 2)
Discrete phase type, state has dimension 2
Reachable states:
State 0: [1, 3]
State 1: [1, 2]
State 2: [1, 1]
State 3: [2, 3]
State 4: [2, 2]
State 5: [2, 1]
State 6: [a, 1]
7 states total
Accepting state index: 6
No trap state
Initial distribution:
    (0:0.666667, 6:0.333333)
Markov chain:
Row 0:
	1 : 1
Row 1:
	2 : 1
Row 2:
	3 : 0.5
	6 : 0.5
Row 3:
	4 : 1
Row 4:
	5 : 1
Row 5:
	6 : 1
Row 6:
	6 : 1
7 edges total
End of information for phase type

=======================================
Checking 5*geometric(0.2)
=======================================
Information for phase type Geometric(0.2)
Discrete phase type, state has dimension 1
Reachable states:
State 0: 1
State 1: a
2 states total
Accepting state index: 1
No trap state
Initial distribution:
    (0:0.2, 1:0.8)
Markov chain:
Row 0:
	0 : 0.2
	1 : 0.8
Row 1:
	1 : 1
1 edges total
End of information for phase type

Information for phase type 5*Geometric(0.2)
Discrete phase type, state has dimension 2
Reachable states:
State 0: [1, 5]
State 1: [1, 4]
State 2: [1, 3]
State 3: [1, 2]
State 4: [1, 1]
State 5: [a, 1]
6 states total
Accepting state index: 5
No trap state
Initial distribution:
    (0:0.2, 5:0.8)
Markov chain:
Row 0:
	1 : 1
Row 1:
	2 : 1
Row 2:
	3 : 1
Row 3:
	4 : 1
Row 4:
	0 : 0.2
	5 : 0.8
Row 5:
	5 : 1
6 edges total
End of information for phase type

=======================================
Checking 5*(geometric(0.5)+geometric(0.2))
=======================================
Information for phase type (Geometric(0.5) + Geometric(0.2))
Discrete phase type, state has dimension 2
Reachable states:
State 0: [1, 0]
State 1: [a, 1]
State 2: [a, a]
3 states total
Accepting state index: 2
No trap state
Initial distribution:
    (0:0.5, 1:0.1, 2:0.4)
Markov chain:
Row 0:
	0 : 0.5
	1 : 0.1
	2 : 0.4
Row 1:
	1 : 0.2
	2 : 0.8
Row 2:
	2 : 1
3 edges total
End of information for phase type

(Natural order)
Information for phase type 5*(Geometric(0.5) + Geometric(0.2))
Discrete phase type, state has dimension 3
Reachable states:
State 0: [[1, 0], 5]
State 1: [[a, 1], 5]
State 2: [[1, 0], 4]
State 3: [[a, 1], 4]
State 4: [[1, 0], 3]
State 5: [[a, 1], 3]
State 6: [[1, 0], 2]
State 7: [[a, 1], 2]
State 8: [[1, 0], 1]
State 9: [[a, 1], 1]
State 10: [[a, a], 1]
11 states total
Accepting state index: 10
No trap state
Initial distribution:
    (0:0.5, 1:0.1, 10:0.4)
Markov chain:
Row 0:
	2 : 1
Row 1:
	3 : 1
Row 2:
	4 : 1
Row 3:
	5 : 1
Row 4:
	6 : 1
Row 5:
	7 : 1
Row 6:
	8 : 1
Row 7:
	9 : 1
Row 8:
	0 : 0.5
	1 : 0.1
	10 : 0.4
Row 9:
	1 : 0.2
	10 : 0.8
Row 10:
	10 : 1
13 edges total
End of information for phase type

(Lexical order)
Information for phase type 5*(Geometric(0.5) + Geometric(0.2))
Discrete phase type, state has dimension 3
Reachable states:
State 0: [[1, 0], 1]
State 1: [[1, 0], 2]
State 2: [[1, 0], 3]
State 3: [[1, 0], 4]
State 4: [[1, 0], 5]
State 5: [[a, 1], 1]
State 6: [[a, 1], 2]
State 7: [[a, 1], 3]
State 8: [[a, 1], 4]
State 9: [[a, 1], 5]
State 10: [[a, a], 1]
11 states total
Accepting state index: 10
No trap state
Initial distribution:
    ([[1, 0], 5]:0.5, [[a, 1], 5]:0.1, [[a, a], 1]:0.4)
Markov chain:
Row 0:
	4 : 0.5
	9 : 0.1
	10 : 0.4
Row 1:
	0 : 1
Row 2:
	1 : 1
Row 3:
	2 : 1
Row 4:
	3 : 1
Row 5:
	9 : 0.2
	10 : 0.8
Row 6:
	5 : 1
Row 7:
	6 : 1
Row 8:
	7 : 1
Row 9:
	8 : 1
Row 10:
	10 : 1
13 edges total
End of information for phase type

